### ISSUE TITLE
What is the difference between torch.mean and caffe2 ReduceMean?

### ISSUE THREAD

**Issue by DenisYarullin** [pytorch/pytorch#39641]
## 🐛 Bug

I manually convert the model from Caffe2 to Pytorch. I built the full architecture of the model in Pytorch using weights from a Caffe2. In Caffe2, the model has a ReduceMean layer, which in Pytorch I replaced with torch.mean. As a result of the replacement, the difference in the calculations turned out to be too large, which does not allow to complete the conversion successfully.

## To Reproduce

1. [input_data.txt](https://github.com/pytorch/pytorch/files/4742801/input_data.txt)
1.
 ```
import numpy as np
import torch
from caffe2.python import workspace, core

data = np.load("input_data.npy")

#caffe2
op_reduce_mean = core.CreateOperator("ReduceMean", ["X_reduce"], ["Y_reduce"], axes=(3,), keepdims=1)
workspace.ResetWorkspace()
workspace.FeedBlob("X_reduce", data)
workspace.RunOperatorOnce(op_reduce_mean)
shape_reduce_mean_caffe2 = workspace.FetchBlob("Y_reduce").shape
data_reduce_mean_caffe2 = workspace.FetchBlob("Y_reduce")
data_reduce_mean_caffe2 = np.array(data_reduce_mean_caffe2, dtype=np.float32).reshape(data_reduce_mean_caffe2.shape)
print(data_reduce_mean_caffe2)  # -0.4089698, -0.5118571, -0.5328341, -0.50671, ... , -0.5756652, -0.38777262, -0.43768662, -0.49657446

#pytorch
torch_data = torch.from_numpy(data)
data_mean_torch = torch.mean(torch_data, dim=(3,), keepdim=True)
print(data_mean_torch)  # -0.4089695, -0.5118583, -0.532835, -0.50670993, ... , -0.57566583, -0.38777304, -0.43768588, -0.49657464

#numpy
data_mean_numpy = np.mean(data, asix=(3,), keepdims=True)  # -0.40896943, -0.5118579, -0.53283495, -0.50670964, ..., -0.5756654, -0.38777274, -0.43768603, -0.49657455
print(data_mean_numpy)
```

## Expected behavior

The same results.

## Environment

 - PyTorch Version (e.g., 1.0): 1.5
 - OS (e.g., Linux): Windows 10
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.7
 - CUDA/cuDNN version: No
 - GPU models and configuration:
 - Any other relevant information:

**Comment by peterjc123** [comment]
Quoted from http://en.wikipedia.org/wiki/Single-precision_floating-point_format

> This gives from 6 to 9 significant decimal digits precision (if a decimal string with at most 6 significant decimal is converted to IEEE 754 single precision and then converted back to the same number of significant decimal, then the final string should match the original; and if an IEEE 754 single precision is converted to a decimal string with at least 9 significant decimal and then converted back to single, then the final number must match the original).

**Comment by DenisYarullin** [comment]
@peterjc123 
You're right. That's my mistake. My BatchNorm2d layer above was in training mode.
